WeatherApi

```diff
@@ -1 +1,41 @@
 
+using Microsoft.Data.SqlClient; 
+using Newtonsoft.Json.Linq; 
+using System; 
+using System.Net; 
+namespace WeatherApi
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            string results;
+            string city;
+            Console.Write("Enter a city- ");
+            city = Console.ReadLine();
+            using (WebClient wc = new WebClient())
+            {
+                string searchtext = "select item.condition from weather.forecast where woeid in (select woeid from geo.places(1) where text='" + city + "') and u = 'c'";         
+
+                results = wc.DownloadString("https://query.weatherapis.com/v1/public/search?q=" + searchtext + "&format=json");
+                dynamic jo = JObject.Parse(results);
+                var items = jo.query.results.channel.item.condition;
+                Audit_weather(city, items.temp, items.text);
```

Comment:
Bad naming conventions. Public methods should follow Pascal case AuditWeather();

This should be done with a Repository pattern or CQRS.

Potentially fire off an AuditWeatherDbCommand and handle it accordingly.

In the AuditWeatherDbCommandHandler use Dapper to perform the insert and avoid SQL injection attacks.

Consider storing database credentials in a Secure Vault with an API and retrieve the password at runtime.